### PR TITLE
feat: add sidebar to the right side

### PR DESCRIPTION
**Pull request by jdjuan**
screenshot-localhost-4200-2018 05 20-13-10-59
screenshot-localhost-4200-2018 05 20-13-10-54

### CODE REVIEW

File: src/app/shared/shared.module.ts [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
-  imports: [CommonModule, MaterialModule],
-  declarations: [
-    FooterComponent,
-    HeaderComponent,
-    MadnessDirective,
-    MenuComponent,
-    ParallaxBigBangDirective,
-    ParallaxHoverDirective,
-  ],
+  imports: [CommonModule, MaterialModule, RouterModule],
```
Comment (luchillo17) [inline src/app/shared/shared.module.ts]: Why a router module in a shared module?

File: src/app/shared/shared.module.ts [hunk 2]
```diff
@@ -1,31 +1,23 @@
...
+  declarations: [FooterComponent, MadnessDirective, MenuComponent, ParallaxBigBangDirective, ParallaxHoverDirective],
```
[the remark this answers is not in the document]
Comment (jdjuan) [reply src/app/shared/shared.module.ts]: To use `routerLink` instead of `href` and avoid the reloading of the website when the route changes

**Comment by jdjuan** [comment]
As to the original question. Well, it turns out that the side menu expects the content of the whole website to be inside of it. Therefore I had to use content projection to push the `router-outlet` and `footer` inside

**Comment by luchillo17** [comment]
I figured as much, i'm just saying neither the Header, Footer & Menu are shared components, i think they should be at the top level along with AppComponent, no need for any share since the app component makes use of them outside of the routes..

**Comment by jdjuan** [comment]
Ah, okay. True.
How would you advice me to locate them?
If I move them to the `app` folder, those `components` will live alongside `modules` which wouldn't be very readable.

How about moving the whole `app module` along with its components to a folder named `root`?

**Comment by luchillo17** [comment]
Hmm, not sure about that, the most common thing to do is to have the app component inside app, then nest components as needed, check this section of angular style guide:
https://angular.io/guide/styleguide#core-feature-module

**Comment by jdjuan** [comment]
Alright, I will add them to a `core` module

File: src/app/home/home-banner/home-banner.component.html [hunk 3]
```diff
@@ -3,4 +3,14 @@
         <h2>{{bigText}}<span class="roboto">.</span></h2>
         <h4>{{subText}}<span class="roboto">.</span></h4>
     </div>
+    <ng-container *ngFor="let image of images">
+        <img
+        alt=""
+        class="ball"
+        [src]='getUrl(image)'
+        [class.bottom]="image.layer === 'bottom'"
+        [class.front]="image.layer === 'front'"
```
Comment (luchillo17) [inline src/app/home/home-banner/home-banner.component.html]: If the class is either `bottom` or `front`, why not using `ngClass` to pass the `image.layer` content as the class?

File: src/app/shared/directives/parallax-scroll.directive.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import { element } from 'protractor';
+import { Directive, ElementRef, OnInit, AfterViewInit } from '@angular/core';
+import { Observable, fromEvent } from 'rxjs';
+import { HostListener } from '@angular/core';
```
Comment (luchillo17) [inline src/app/shared/directives/parallax-scroll.directive.ts]: Imports lack order.

File: src/app/shared/shared.module.ts [hunk 5]
```diff
@@ -1,22 +1,25 @@
...
 import { ParallaxHoverDirective } from './directives/parallax-hover.directive';
-import { RouterModule } from '@angular/router';
+import { ParallaxScrollDirective } from './directives/parallax-scroll.directive';
+import { FooterComponent } from './footer/footer.component';
+import { MenuComponent } from './menu/menu.component';
 
 @NgModule({
   imports: [CommonModule, MaterialModule, RouterModule],
-  declarations: [FooterComponent, MadnessDirective, MenuComponent, ParallaxBigBangDirective, ParallaxHoverDirective],
+  declarations: [FooterComponent, MadnessDirective, MenuComponent, ParallaxBigBangDirective, ParallaxScrollDirective, ParallaxHoverDirective],
```
Comment (luchillo17) [inline src/app/shared/shared.module.ts]: Is the parallax directives actually shared or only for home page?

**Pull request merged**